app/uk/gov/hmrc/lisaapi/controllers/BonusPaymentController.scala

```diff
@@ -219,82 +237,54 @@ class BonusPaymentController extends LisaController with LisaConstants {
-
-        Forbidden(Json.toJson(ErrorBonusClaimError))
-      case RequestBonusPaymentAccountClosed =>
-        auditFailure(lisaManager, accountId, req, ErrorAccountAlreadyClosedOrVoid.errorCode)
-        LisaMetrics.incrementMetrics(startTime, FORBIDDEN, LisaMetricKeys.BONUS_PAYMENT)
-
-        Forbidden(Json.toJson(ErrorAccountAlreadyClosedOrVoid))
-      case RequestBonusPaymentAccountCancelled =>
-        auditFailure(lisaManager, accountId, req, ErrorAccountAlreadyClosedOrVoid.errorCode)
-        LisaMetrics.incrementMetrics(startTime, FORBIDDEN, LisaMetricKeys.BONUS_PAYMENT)
-
-        Forbidden(Json.toJson(ErrorAccountAlreadyClosedOrVoid))
-      case RequestBonusPaymentAccountVoid =>
-        auditFailure(lisaManager, accountId, req, ErrorAccountAlreadyClosedOrVoid.errorCode)
-        LisaMetrics.incrementMetrics(startTime, FORBIDDEN, LisaMetricKeys.BONUS_PAYMENT)
-
-        Forbidden(Json.toJson(ErrorAccountAlreadyClosedOrVoid))
-      case RequestBonusPaymentSupersededAmountMismatch =>
-        auditFailure(lisaManager, accountId, req, ErrorBonusSupersededAmountMismatch.errorCode)
-        LisaMetrics.incrementMetrics(startTime, FORBIDDEN, LisaMetricKeys.BONUS_PAYMENT)
-
-        Forbidden(Json.toJson(ErrorBonusSupersededAmountMismatch))
-      case RequestBonusPaymentSupersededOutcomeError =>
-        auditFailure(lisaManager, accountId, req, ErrorBonusSupersededOutcomeError.errorCode)
-        LisaMetrics.incrementMetrics(startTime, FORBIDDEN, LisaMetricKeys.BONUS_PAYMENT)
-
-        Forbidden(Json.toJson(ErrorBonusSupersededOutcomeError))
-      case RequestBonusPaymentNoSubscriptions =>
-        auditFailure(lisaManager, accountId, req, ErrorNoSubscriptions.errorCode)
-        LisaMetrics.incrementMetrics(startTime, FORBIDDEN, LisaMetricKeys.BONUS_PAYMENT)
-
-        Forbidden(Json.toJson(ErrorNoSubscriptions))
-      case RequestBonusPaymentAccountNotFound =>
-        auditFailure(lisaManager, accountId, req, ErrorAccountNotFound.errorCode)
-        LisaMetrics.incrementMetrics(startTime, NOT_FOUND, LisaMetricKeys.BONUS_PAYMENT)
-
-        NotFound(Json.toJson(ErrorAccountNotFound))
-      case RequestBonusPaymentLifeEventNotFound =>
-        auditFailure(lisaManager, accountId, req, ErrorLifeEventIdNotFound.errorCode)
-        LisaMetrics.incrementMetrics(startTime, NOT_FOUND, LisaMetricKeys.BONUS_PAYMENT)
-
-        NotFound(Json.toJson(ErrorLifeEventIdNotFound))
+    val response: ErrorResponse = errorResponse match {
       case e: RequestBonusPaymentClaimAlreadyExists =>
-        val error = ErrorBonusClaimAlreadyExists(e.transactionId)
-
-        auditFailure(lisaManager, accountId, req, error.errorCode)
-        LisaMetrics.incrementMetrics(startTime, CONFLICT, LisaMetricKeys.BONUS_PAYMENT)
-
-        Conflict(Json.toJson(error))
+        getAPIVersionFromRequest(request) match {
+          case Some(VERSION_1) => ErrorInternalServerError
```

Comment:
Could each of these cases covered by the `requestBonusErrorsV1.getOrElse(errorResponse, ErrorInternalServerError)` section a few lines below?

Reply:
Perhaps but it wouldn't be as simple as that as this case and the one below it are handling classes with ids. The catch all (.getOrElse) is merely handling simple objects without any properties.